archunit-junit/junit5/engine/src/test/java/com/tngtech/archunit/junit/ArchUnitTestEngineTest.java

```diff
@@ -514,6 +589,57 @@ void complex_tags() {
                             TestTag.create("method-tag"));
         }
 
+        @Test
+        void complex_meta_tags() {
+            EngineDiscoveryTestRequest discoveryRequest = new EngineDiscoveryTestRequest().withClass(ComplexMetaTags.class);
+
+            TestDescriptor descriptor = testEngine.discover(discoveryRequest, engineId);
+
+            Map<UniqueId, Set<TestTag>> tagsById = new HashMap<>();
+            descriptor.accept(d -> tagsById.put(d.getUniqueId(), d.getTags()));
+
+            assertThat(getTagsForIdEndingIn(ComplexMetaTags.class.getSimpleName(), tagsById))
+                    .containsOnly(TestTag.create("library-tag"));
+
+            assertThat(getTagsForIdEndingIn(TestClassWithMetaTag.class.getSimpleName(), tagsById))
```

Comment:
Similarly to https://github.com/TNG/ArchUnit/pull/287#discussion_r365862755, the assertions only appear the same because I used the same `@ArchTag`s in `ComplexMetaTags` as `ComplexTags`

Reply:
I have renamed the `@ArchTag`s in the new test classes to distinguish them from the existing test classes, so that the assertions no longer appear the same